text/docs-code-contributions.md

```diff
@@ -0,0 +1,59 @@
+- Start Date: 2019-02-07)
+- RFC PR: 
+- Gatsby Issue:
+
+# Summary
+
+Reorganize the contributor documentation to aid in Gatsby core and docs development.
+
+# Motivation
+
+The Gatsby.js docs, while wonderful, are light on coding instructions for contributors. With dedicated sections in the Contributing docs, it will be easier to learn how to clone the repository, where to find certain things like docs or plugins, how to run tests, and insider infrastructure knowledge required to submit a successful pull request on the first try.
+
+A lot of this information already lives on the "How to Contribute" page, but it's buried. It would help discoverability if this information lived on separate pages.
+
+# Detailed design
+
+I propose that we add some pages to the existing contributor documentation and move some content around. Here is an IA diagram that (briefly) reflects the current structure and what I think we should add (open for discussion, of course):
+
+Contributing
+- Community
+    - Pair Programming Sessions
+    - Where to participate in the community (this was in RFCs, perhaps it could be reintroduced and content moved from "How to contribute"?)
+    - How to run a Gatsby workshop*
+    - How to pitch Gatsby*
+- Code of Conduct
+- Gatsby Style Guide
+- How to contribute
+    - How to file an issue
+    - How to label an issue
+    - Free swag for contributors
+    - Contributing to the documentation
+        - Docs templates
+    - Contributing to the repo
+        - page includes moved content from "How to contribute" incl. starter library, website, everything that isn't docs (with a mention of plugins section).
+    - Setting up test environments (Wordpress, Contentful, etc.)
+- RFC process
+- Submit to Site Showcase
+- Submit to Creator Showcase
+- Submit to Starter Library
+- Submit to Plugin Library
+
+# Drawbacks
+
+Why should we *not* do this? Please consider:
+
+- we will need some redirects so people who have bookmarked things don't get 404s
+- there will be a user adjustment required to learn the new locations of things
+
+# Alternatives
+
+Adding more top-level docs instead of reorganizing them. It's a pretty large section though, and it seems like it could be reorganized.
```

Comment:
Another option here is to separate out contributing docs into a separate section. @amberleyromo, @shannonbux, and I have discussed the potential grouping of docs by the task someone wants to accomplish:

1. I want to build a website using Gatsby (e.g. the tutorial)
2. I want to build something _for_ or _in_ Gatsby (e.g. plugins, themes, contribution)
3. I need an API reference

If "Contributing" became its own top-level page (e.g. `https://gatsbyjs.org/contributing`) it could have its own sidebar nav. The docs could then have a single "Contribute to Gatsby" nav item in the left nav/in the footer that takes people to this section.

My worry for a while now has been that there's so much info in the docs that we can't put it all in the same place. We need contextual grouping so people are able to focus on one set of concerns at a time vs. "everything that's possible in Gatsby is here".

Reply:
I like the idea of a separate "Contributing" section, especially when I look at the other top-level sections (tutorial, plugins, features, etc.). Redirects and some links from the docs would help ease the transition.

One thing that's currently confusing about the docs is how "Documentation" encompasses everything in the /docs/ part of the URL schema, but visually it is its own expandable section:
<img width="244" alt="Gatsby sidebar" src="https://user-images.githubusercontent.com/1045233/52513260-806e3e80-2bbe-11e9-8072-3aef17e7ab16.png">

Giving Contributing its own sidebar would make it less overwhelming. I also like the idea of grouping things by tasks–I can work on an update to this RFC that incorporates those ideas.

Reply:
Oh, I hadn't read this before I left this earlier comment https://github.com/gatsbyjs/rfcs/pull/29/files#r256533979. In sum, I agree with making this a new page with it's own sidebar!